### PR TITLE
Update rubocop and rubocop-rspec

### PR DESCRIPTION
**Pull request by nschonni**
- Update to the latest versions
- Autofixed new rules
- Regenerated TODO files for some other fixed items

### CODE REVIEW

File: app/lib/importer/base_importer.rb [hunk 1]
```diff
@@ -68,8 +68,8 @@ def clean_up!
 
   protected
 
-  def in_work_unit(*args, &block)
-    work_unit = Concurrent::Promises.future_on(@executor, *args, &block)
+  def in_work_unit(...)
+    work_unit = Concurrent::Promises.future_on(@executor, ...)
```
Comment (nschonni) [inline app/lib/importer/base_importer.rb:72]: This originally had a bad fix from https://github.com/rubocop/rubocop/issues/12087, but manually changed fixed it

**Comment by nschonni** [comment]
The new RSpec rule might not be fully safe autofix https://github.com/rubocop/rubocop-rspec/issues/1677, but after addressing the first bad autofix, the specs still seem happy

**Review by renchap — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
I'm not sure we want to update to something that we know have unsafe autofix, but the actual code changes are fine.

**Comment by mjankowski** [comment]
> I'm not sure we want to update to something that we know have unsafe autofix, but the actual code changes are fine.

I have a process question for these style/lint version bump PRs ... is your preference to have either:

- A PR in this style, where the version bump, the todo regeneration, and all the required changes are all in one big PR (and maybe even one larger commit)
- More isolated per-change PRs/commits leading up to the version bump/regen (ie for this particular one we'd have a PR for the message chain change, a PR for the argument forwarding; and then only after those are merged, a final PR doing the actual version bump and todo regen if necessary)

My suspicision is that the latter might be easer to review, would love to confirm.

**Comment by ClearlyClaire** [comment]
I can see both ways having their own advantages:
- when the changes are purely automatic, having the rule changes before or at the same time as the code changes can help forks which can just discard the automated changes and re-run the autofixer
- having separate PRs can on the other hand be very slightly easier to review in the main project, and are a bit cleaner in the commit history due to squash-merging

I do not feel very strongly about either solution, tbh.

**Comment by renchap** [comment]
`rubocop-rspec` [v2.23.2](https://github.com/rubocop/rubocop-rspec/releases/tag/v2.23.2) fixes the issue with `RSpec/ReceiveMessages` autofix, could you update your PR?

Also there is a new Rubocop version as well.

**Comment by nschonni** [comment]
I think https://github.com/rubocop/rubocop/issues/12111 needs to be released, as I get an error running with the newest rubocop

**Comment by renchap** [comment]
Looks good, I will let @ClearlyClaire merge!

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**